conans/client/loader.py

```diff
@@ -76,6 +76,7 @@ def load_conan(self, conan_file_path, consumer=False):
             result = self._create_check_conan(loaded, consumer)
             if consumer:
                 result.options.initialize_upstream(self._options)
+	    type(result).conanfile_directory = property(lambda self, _path=conan_file_path: os.path.dirname(_path))
```

Comment:
I'll cherry pick out this commit and create a separate pull request.  We can discuss the best way to handle it in there.